src/core/ddsi/tests/CMakeLists.txt

```diff
@@ -12,6 +12,7 @@
 include(CUnit)
 
 set(ddsi_test_sources
+    "locators.c"
```

Comment:
Haha! Either I'm losing my touch or I am not a fan "do as I say, not as I do" 🙂

FIY: I just discovered that a ROS 2 CI on #558 failed because this particular test pulls in an OpenSSL header file, which then isn't found on Windows & macOS in the ROS 2 CI system (the CMake file doesn't declare a dependency, so it is ok that that happens; Linux is fine by the way).

There is no reason why that dependency should exist, so I'm quickly retesting with an additional commit on #558 (https://github.com/eclipse-cyclonedds/cyclonedds/pull/558/commits/65db8fb2425c9a61592444b4ebd6ed3eacd59e09) and I intend to squash that commit into this one before merging this PR.

Reply:
With that change in, ROS 2 CI passes. Instead of squashing it, I've inserted it just before the change that introduced the problem ("Remove IPv4/IPv6-specific code in adding peers"). Also added a missing `extern "C"`.